### DISCUSSION TITLE
SSL_CERT UNKNOWN mydomain.com: /tmp is not writable

### DISCUSSION THREAD

**Discussion by jainjine** [matteocorti/check_ssl_cert#332] [General]
Hi,

I am using the check_ssl_cert  ( v2.9.0 ) I am getting this error "SSL_CERT UNKNOWN mydomain.com: /tmp is not writable"

could you pls help me on this?

**Comment by matteocorti** [comment]
Hi

the plugin needs to create some temporary files. These are usually created in the directory specified by the environment variable ```$TMPDIR``` (see [TMPDIR on Wikipedia](https://en.wikipedia.org/wiki/TMPDIR)). In your case, either ```$TMPDIR``` specifies ```/tmp``` or is empty (the plugin uses ```/tmp``` per default) and ```/tmp``` is not writable by the user running the script.

You can

- Check If ```$TMPDIR``` is correctly set and change it if not
- or correct the permissions of ```/tmp```
- or specify another directory where to create the temporary files with the ```--temp``` command line option (e.g., ```./check_ssl_cert -H example.com --temp /some/other/directory```

Matteo

**Reply by jainjine** [reply]
Hi

My /tmp permission  is "drwxrwxrwt. 9 root root 283 Oct 23 10:50 /tmp" could you pls tell me which permission which I need to change for /tmp ?

**Comment by matteocorti** [comment]
Can you please provide the output of

```
echo $TMPDIR; test -w $TMPDIR; echo $?; ls -ld $TMPDIR
```

**Comment by jainjine** [comment]
dr-xr-x---. 13 root root 4096 Oct 23 11:45

**Comment by matteocorti** [comment]
This is not the output of the command I specified, there should be three lines. Something like:

```
/var/folders/m3/9pmctdmn4xjd7h06s3b6fl0m0000gn/T/
0
drwx------@ 152 corti  staff  4864 Oct 25 09:18 /var/folders/m3/9pmctdmn4xjd7h06s3b6fl0m0000gn/T/
```

Anyway if the output you pasted refers to ```${TMPDIR}``` then the permissions are clearly wrong. It would be interesting to know what is ```${TMPDIR}``` in your case (according to the plugin ```/tmp``` but according to your post the permissions of ```/tmp``` are OK and different from your last answer.

Try with

```
sudo chmod 1777 ${TMPDIR}
```

**Reply by jainjine** [reply]
I have changed the permission to 1777 but still the same. It can be work without using TMP?

**Comment by matteocorti** [comment]
Of course, as I said at the beginning you can specify another directory where to create the temporary files with the ```--temp``` command line option (e.g., ```./check_ssl_cert -H example.com --temp /some/other/directory```)

**Reply by jainjine** [reply]
where I need to create this ? in commands.cfg file?

**Comment by matteocorti** [comment]
Sorry but create what? Yes command line options are specified in the ```commands.cfg``` file. But I cannot provide help for Nagios, you should ask a support forum for the tool you are using.

**Reply by jainjine** [reply]
what I understood here first I have to create a temp folder in my Nagios Monitoring Server and that path is needed to specify in commands.cfg file correct?

**Reply by matteocorti** [reply]
You don't need to create a folder, the plugin just need a place where it can write a temporary files. This should be your $TMPDIR directory. If you really want create a folder and then use the command line option to specify which folder you created.

**Reply by jainjine** [reply]
Ok now I understood...I appreciate your valuable time which you spent here with me..

**Reply by jainjine** [reply]
Hi, The issue has resolved now.. because of the SELINUX is not disabled, after disabled its working fine..

Thanks for your time to spent here.
